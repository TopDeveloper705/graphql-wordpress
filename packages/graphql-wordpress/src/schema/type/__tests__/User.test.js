import UserType from 'type/User';

// eslint-disable-next-line no-underscore-dangle
const fields = UserType._typeConfig.fields;

describe('Test User type', () => {
  test('Test name', () => {
    expect(UserType.name).toMatchSnapshot();
  });

  test('Test fields', () => {
    expect(Object.keys(UserType.getFields())).toMatchSnapshot();
  });

  test('Test resolve avatars', () => {
    const avatars = fields.avatarUrls.resolve({
      avatar_urls: {
        thumbnail: 'https://scott.com/avi.jpg',
      },
    });
    expect(avatars).toMatchSnapshot();
  });
});
