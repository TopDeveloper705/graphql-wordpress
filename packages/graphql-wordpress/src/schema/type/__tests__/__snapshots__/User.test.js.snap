// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User type Test fields 1`] = `
Array [
  "id",
  "name",
  "description",
  "link",
  "slug",
  "avatarUrls",
]
`;

exports[`Test User type Test name 1`] = `"User"`;

exports[`Test User type Test resolve avatars 1`] = `
Array [
  Object {
    "size": "thumbnail",
    "url": "https://scott.com/avi.jpg",
  },
]
`;
